src/compiler/checker.ts

```diff
@@ -6730,21 +6783,12 @@ namespace ts {
                 }
                 return unknownType;
             }
+
+            // TODO: Why is this line here?

```

Comment:
It's to cache the result, so that if we ask for it again, we have it already. The odd thing is that we do not check resolvedSymbol earlier in this function. So it must be that we check it somewhere else, but usually the check and the caching are done in the same function.